zilencer/views.py

```diff
@@ -153,23 +154,35 @@ def register_remote_server(
 def register_remote_push_device(
     request: HttpRequest,
     entity: Union[UserProfile, RemoteZulipServer],
-    user_id: int = REQ(json_validator=check_int),
+    user_id: Optional[int] = REQ(json_validator=check_int, default=None),
+    user_uuid: Optional[str] = REQ(default=None),
     token: str = REQ(),
     token_kind: int = REQ(json_validator=check_int),
     ios_app_id: Optional[str] = None,
 ) -> HttpResponse:
     server = validate_bouncer_token_request(entity, token, token_kind)
 
+    if not (user_id or user_uuid):
+        raise JsonableError(_("Missing user_id or user_uuid"))
+    if user_id and user_uuid:
+        raise JsonableError(_("Specify only one of user_id or user_uuid"))
+
+    if user_id:
+        kwargs: Dict[str, Union[str, int]] = {"user_id": user_id}
+    else:
+        assert user_uuid
+        kwargs = {"user_uuid": user_uuid}
```

Comment:
I don't know that I find this more understandable than passing both:
```py3
            RemotePushDeviceToken.objects.create(
                user_id=user_id,
                user_uuid=user_uuid,
                server=server,
                kind=token_kind,
                token=token,
                ios_app_id=ios_app_id,
                # last_updated is to be renamed to date_created.
                last_updated=timezone.now(),
            )
```

I don't think this construction adds any type safety.

Reply:
I think the idea might be that we don't want to record the `user_id` field at all if the other field is available, so that servers immediately get improved privacy of not sharing user IDs with us.

Reply:
That doesn't make sense.  This codepath is on the server, and is after a check for `if user_id and user_uuid: raise JsonableError(_("Specify only one of user_id or user_uuid"))`

Reply:
Ahh, yes, I hadn't clicked into the context and thought this was a different code path.  I agree that what you propose would be an improvement here, since then we're explicitly setting the other values to `None`.

Reply:
Changed this, and also changed the falsey checks to have `is not None`, since otherwise `0` would produce unexpected results.

Reply:
@timabbott Hmm, the idea of the checks is that `0` is  not really a valid value and we should treat it like `None`?